opentelemetry-exporter-cloud-trace/tests/test_integration_cloud_trace_exporter.py

```diff
@@ -0,0 +1,88 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import socket
+import subprocess
+import unittest
+from time import sleep
+
+import grpc
+from google.cloud.trace_v2 import TraceServiceClient
+from google.cloud.trace_v2.gapic.transports import trace_service_grpc_transport
+from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace.export import Span, SpanExportResult
+from opentelemetry.trace import SpanContext, SpanKind
+
+
+class BaseExporterIntegrationTest(unittest.TestCase):
+    def setUp(self):
+        # Find a free port to spin up our server at.
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.bind(("localhost", 0))
+        self.address = "localhost:" + str(sock.getsockname()[1])
+        sock.close()
+
+        # Start the mock server.
+        args = "mock_server-x64-linux -address=" + self.address
+        self.mock_server_process = subprocess.Popen(args, shell=True)
+
+    def tearDown(self):
+        self.mock_server_process.kill()
+
+
+class TestCloudTraceSpanExporter(BaseExporterIntegrationTest):
+    def test_export(self):
+        project_id = "TEST-PROJECT"
+        trace_id = "6e0c63257de34c92bf9efcd03927272e"
+        span_id = "95bb5edabd45950f"
+
+        # Create span and associated data.
+        resource_info = Resource(
+            {
+                "cloud.account.id": 123,
+                "host.id": "host",
+                "cloud.zone": "US",
+                "cloud.provider": "gcp",
+                "gcp.resource_type": "gce_instance",
+            }
+        )
+        span = Span(
+            name="span_name",
+            context=SpanContext(
+                trace_id=int(trace_id, 16),
+                span_id=int(span_id, 16),
+                is_remote=False,
+            ),
+            parent=None,
+            kind=SpanKind.INTERNAL,
+            resource=resource_info,
+            attributes={"attr_key": "attr_value"},
+        )
+        span.start()
+        sleep(0.1)
+        span.end()
+        span_data = [span]
```

Comment:
Instead of doing `span.start()` `span.end()` can you just manually set the `end_time` and `start_time`

Reply:
Yep, done

Reply:
You should also do `span.end(...)` in that case. 

I'm still concerned about this approach since `.start()` `.end()` do a lot of other stuff like interacting with the `span_processor`. Currently this is a no-op, but in the future this may change (e.g. SpanProcessor becomes an abstract class the throws if it gets initialized + called). I still think we should set `_start_time` `_end_time` manually and just tell pylint to not complain. @aabmass any thoughts?

Reply:
Gotcha, if `.start()` do unwanted things then I agree that setting the fields would be better, so I've made this change.

Reply:
If you want to completely get rid of any side effects from `Span`, you can use a `Mock()`. You might be able to just replace your call to `Span()` above with `Mock()` (same exact arguments) e.g.

```py
span = Mock(
    spec=Span,
    start_time=whenever,
    end_time=whenver + 10 ** 9,

    name="span_name",
    context=SpanContext(
        trace_id=int(trace_id, 16), span_id=int(span_id, 16), is_remote=False,
    ),
    parent=None,
    kind=SpanKind.INTERNAL,
    resource=resource_info,
    attributes={"attr_key": "attr_value"},
)
```

The `spec=Span` kwarg to Mock  makes it raise an error if the Span is misused also

Reply:
I tried using `mock.Mock()` and it gives `return "{:032x}".format(trace_id)` `TypeError: unsupported format string passed to Mock.__format__` :(